src/configuration.ts

```diff
@@ -120,8 +120,83 @@ export function loadConfigurationFromPath(configFilePath: string) {
         let fileData = fs.readFileSync(configFilePath, "utf8");
         // remove BOM if present
         fileData = fileData.replace(/^\uFEFF/, "");
-        return JSON.parse(fileData);
+        let config = JSON.parse(fileData) as IConfigFile;
+
+        let parentConfigNames = config.extends;
+        if (parentConfigNames) {
+            const configFileDir = path.dirname(configFilePath);
+            if (Array.isArray(parentConfigNames)) {
```

Comment:
Let's do

```
if (typeof parentConfigNames === "string") {
    parentConfigNames = [parentConfigNames];
}
```

And then below, we can just handle the array case and not have to duplicate logic